### DISCUSSION TITLE
mysql_query removed on php7

### DISCUSSION THREAD

**Discussion by ericc-dream** [e107inc/e107#5261] [Q&A Updating] [answered]
Hi All 

I'm using e107 since a long time, even before version 1.
I upgrade it some time to time because each time I have troubles. 
Actually, my site is (was!) running on version 2.2.1 with php7.2

I need to change of hosting service, but all offer propose only 8.1 or higher ... so I try to upgrade to 2.3.3 
I reinstalled everything on my "test" web server  with the same configuration (php7.2 then) and on this one I can check the php errors in /var/log/apache2/error.php 
To my surprise, I see a lot of error like :
[Thu May 16 20:52:13.916322 2024] [php7:notice] [pid 1760] [client 192.168.28.100:41756] PHP Fatal error:  Uncaught Error: Call to undefined function **mysql_query()** in /home/ericc/www/ericc/e107_plugins/content/content_update_check.php:51\nStack trace:\n#0 /home/ericc/www/ericc/e107_admin/update_routines.php(429): update_content_07(false)\n#1 /home/ericc/www/ericc/e107_admin/boot.php(45): update_check()\n#2 /home/ericc/www/ericc/e107_admin/admin.php(72): require_once('/home/ericc/www...')\n#3 {main}\n  thrown in /home/ericc/www/ericc/e107_plugins/content/content_update_check.php on line 51, referer: http://192.168.28.114/ericc/e107_admin/admin.php

Some seems to be commented out, but I still found 89 still used ! 
According to [https://www.php.net/manual/en/function.mysql-query](https://www.php.net/manual/en/function.mysql-query) this function is removed from 7.0 already 

Are you aware ?

I try to upgrade to 2.3.3 by copying the content of archive e107_v2.x_to_2.3.3_upgrade.tar.xz, but nothing seems to happen ... usually we have to run something !? an update of the DB or whatever ! 
I tried an upgrade to 2.3.2 and I had to take some action ... so why not when going straight to 2.3.3 ?

I checked in "e107_2.2.1_full", "e107_2.3.3_full", "e107_v2.x_to_2.3.2_upgrade" and "e107_v2.x_to_2.3.3_upgrade" and in all I found some call to "mysql_query()" ... 

Can I replace all instance of 
`mysql_query($qry)` 
by 
`$sql->gen($qry)`  ? 

Can't it be a problem in the future when you will correct it ?

**Comment by rica-carv** [comment, marked as answer]
@ericc-dream It's kind of weird, since i'm using e107 v2.3.3 on a server with PHP 7.4, and i got no errors regarding the mysql-query

I've checked along the v2.3.3 code, and all mysql-query code is commented out, so for sure v2.3.3 doesn't use mysql-query....

Probably you have old code in your site, so my advice is you to make a v2.3.3 fresh install, from scratch....
Furthermore, it seems that error comes from a plugin: content. It's not core plugin, so it might be outdated....

**Reply by tgtje** [reply]
@ericc-dream as @rica-carv already mentioned : content plugin is not part of v. 2 (lastdate in core system at v 1.0.4, not maintained anymore).
Functions where incorporated/rewritten into the v2 system  (pages, books, chapters etc..).
I know import plugin can handle some stuff (eg kind of convert old into new), but has been a while .. i do remember that pages created with old tiny and now imported could upset styling issues like breaks etc. in the imported pages.
Give it a try (after backups made). > not installed but present / do install and check it out

**Reply by Jimmi08** [reply]
The latest version of the content plugin:   https://github.com/Jimako-e107-plugins/e107-old-plugins/tree/master/content_fixed 

Nobody needed it, nobody asked for it, nobody wanted to help.  Too late now. 

@ericc-dream  you can remove that line or file at all, it was for update check in core and it is not anymore in core.

**Reply by rica-carv** [reply]
@Jimmi08 I might give a look at it, i'll probably need that plugin....

**Comment by ericc-dream** [comment]
Indeed, I still have some old files from version 1 or even older, in my site 
Yesterday, after writing the message, I restart from the beginning and update my site to 2.3.2 
As expected, I had to click on several buttons to update some files, the DB, some parameter in the forum and I don't remember what else.
Don't understand why this doesn't happen when I upgrade to 2.3.3 directly or indirectly 

Then, I saw a message recommending to use the "file inspector" (not sure of the name, but you probably see what I mean) what I did (I never saw that before and so never used it). It found hundred of old files ! That I removed manually ... 
And, it went too far (or I made a mistake) , as after that I couldn't access to the admin side anymore. 

I will start over once more and remove old plugin that I don't use or are more or less broken 

I did a "grep" on e107_2.3.3_full, and I saw a few lines containing "mysql_query" which doesn't seems to commented out 
I checked this morning, and they are in bigger block (/* ...*/) so all is fine, apologies for that

**Comment by ericc-dream** [comment]
OK, I solved most of my problems 
I removed the plugin "content", because, in fact, I wasn't using it and it's redundant with "pages/menu" , at least for my usage.
Now my main problem is that my site is in French and the language pack is completely out of date , with a lot of translations missing.
With php7, when a constant wasn't defined, or it was ignoring it (in a 'if' structure by example) , or was displaying its name.
with php8, it stop rendering the page ... and sometime nothing at all is displayed ... 
I removed also all old files, more than 1000 ... and it was a lot of work !

**Reply by tgtje** [reply]
@ericc-dream  language go here https://github.com/e107translations/French  
Although that one  is still not fully updated,
A newer pull request is commited,  and you can find it [here](https://github.com/tgtje/French) (waiting for merging).
Click on "code', dowl zip, _extract_: overwrite the folders complete (as they carry identical names (when not being renamed in your system!!) its a copy and paste method. (When in doubt : always backup your files first).

**Comment by ericc-dream** [comment]
I used the first one but it's 4 years old with quite some missing, some are blocking the site or the admin page to load
I will try yours ... if it's OK, it will spare me a lot of works
Thanks

**Comment by ericc-dream** [comment]
@tgtje 

> PHP Parse error:  syntax error, unexpected identifier "rel", expecting ")" in /home/ericc/www/ericc/e107_languages/French/admin/lan_admin.php on line 424

you have put double quote around 'rel' 


> PHP Parse error:  syntax error, unexpected identifier "Version" in /home/ericc/www/ericc/e107_languages/French/admin/help/cpage.php on line 18, referer: http://192.168.28.114/ericc/e107_admin/admin.php

Not sure on this one ! 
There is a line saying:
> Demo Version, Only convert first 5 files at one time, Please buy now!

Did you used a tool to translate the files ?


> PHP Fatal error:  Uncaught Error: Undefined constant "PREVIEWTHEMENAME" in /home/ericc/www/ericc/e107_languages/French/admin/lan_theme.php:12

in your version, you have
`define("TPVLAN_1", "Vous pr&eacute;visualisez le th&egrave;me <b>&apos;".PREVIEWTHEMENAME."&apos;</b>. `
while in the English version :
`define("TPVLAN_1", "You are looking at a preview of the <b>{PREVIEWTHEMENAME}</b> theme.`
look like the translation went too far :-)



Then 2 specifics for me , I guess 
> PHP Fatal error:  Uncaught Error: Undefined constant "LAN_FORUM_3098" in /home/ericc/www/ericc/e107_themes/white-n-black/templates/forum/newforumposts_menu_template.php:25

I have this plugin displayed on the front page but strangely enough, in the admin side, it's not installed !!
This constant "LAN_FORUM_3098" and the next one "LAN_FORUM_3099" aren't present in the English version neither 



> PHP Fatal error:  Uncaught Error: Undefined constant "ADLAN_115" in /home/ericc/www/ericc/e107_plugins/chatbox2_menu/e_status.php:5

I'm still using the very old chatbox2, because I hacked it a lot and corrected it for every version.Probably not acceptable with the current standard but it work very well ... for me and I don't plan to change. I probably need to rework it, again, to use its own languages files.

That all for the moment ... I don't think that I have tried everything but 80% already

**Reply by Jimmi08** [reply]
Use PHP 7

**Reply by rica-carv** [reply]
@ericc-dream You must have really something there heavilly bugged up....

For instance:
 
> PHP Parse error:  syntax error, unexpected identifier "rel", expecting ")" in /home/ericc/www/ericc/e107_languages/French/admin/lan_admin.php on line 424
> 
[This file](https://github.com/e107translations/French/blob/master/e107_languages/French/admin/lan_admin.php) has only 415 lines, how can it have an error on line 424???

> PHP Parse error:  syntax error, unexpected identifier "Version" in /home/ericc/www/ericc/e107_languages/French/admin/help/cpage.php on line 18, referer: http://192.168.28.114/ericc/e107_admin/admin.php
> 
[This file](https://github.com/e107translations/French/blob/master/e107_languages/French/admin/help/cpage.php) has only 17 lines, how can it have an error on line 18???

> Demo Version, Only convert first 5 files at one time, Please buy now!
>
Sorry, but i don't see that text in any line...

For the rest of the errors, you could use the Language area in the admin area, it can correct and check language files against the english lan files, There's no need to do it by hand, it's a lot easier to use the language area in admin menu to edit and update the french lan files....

**Reply by ericc-dream** [reply]
@rica-carv 
I'm using the version of @tgtje ! 
the file you refer is 4 years old and miss several constant 

> @ericc-dream language go here https://github.com/e107translations/French
Although that one is still not fully updated,
A newer pull request is commited, and you can find it [here](https://github.com/tgtje/French) (waiting for merging).
Click on "code', dowl zip, extract: overwrite the folders complete (as they carry identical names (when not being renamed in your system!!) its a copy and paste method. (When in doubt : always backup your files first).

**Reply by tgtje** [reply]
@ericc-dream you are correct : the demo line belongs to deepL converter. 
Since i am not French some parts i  used from it (trial mode).. you can delete the lines..
I already corrected them in the commit ( pages :admin_log.php; administrator.php;cache php and cpage.php all in help/. Thank you !  TVPLAN 1 (thought i corrected that ) will do so again;  also the "rel  lan_admin (edit : all done)

Do not know the theme myself so likely old stuff... 

ADLAN 115 **no longer in use** >  // define("ADLAN_115", "Chatbox posts");
just info new lan_admin has 436 lines (updated 2019>2024)

edit : trying to help out, since translators are hard to find thesedays.. Volunteers ? welcome .;-)

**Reply by Jimmi08** [reply]
@tgtje deepl is enough for file strings :)  
if you want to have your PR merged, ask @Moc on gitter or ask to add you as a language maintainer.  You still need to create release... 

@ericc-dream 
black and white theme is for PHP 7 only. 
https://github.com/Jimako-e107-themes/e107-old-themes/tree/main/black-and-white

**Reply by tgtje** [reply]
@Jimmi08 thank you, i normally ask @Moc , : made a commit, informed, but closed that one. The newer is younger.
deepL is quite good, better than G. mostly ...quick online trans : danngg demo line.. :-(  got through..
The LAN_FORUM_3098 /3099 ???? think a personal added lan, in at least 12 years never seen...

white-n-black theme : also a private build ? (erric's one)   as b-n-w is another name > (one from around 2008 also had such name./ v 7.15  3 column))

**Comment by ericc-dream** [comment]
@Jimmi08 
As far as I remember, I wanted a dark theme, and for v2 there are less and less available themes. 
So I took your "Black & White" theme and "revert" it (switch the colors) , hence the "White N Black" name , plus done some personal enhancement and since I correct it for each version to keep it working ... in the same way that I'm correcting plugin "chatbox2" 
And honestly, there are not a lot to correct to make it work for php8

**Reply by Jimmi08** [reply]
There is in fact a lot of new themes, they are just not uploaded on e107.org .   

What is the correct way if you use the old theme from that list:
- create your own repo, upload it there, and post your fixes there.  

From v 2.3.3 (or 2.3.x) there is a new way how themes work and I will not support legacy themes anymore. 

I asked for the only thing to make new themes - to find the way how to connect them directly with github repos...  to be able to download and update them without uploading them on e107.org.   It is possible (see languages pack).  

By the way, there is a nice new dark theme from Fizi (select gaming theme).      

`https://www.e107sk.com/demo/`

**Comment by ericc-dream** [comment]
@tgtje 
I found another mistake 
> PHP Parse error:  syntax error, unexpected identifier "Hero", expecting ")" in /home/ericc/www/ericc/e107_plugins/hero/languages/French/French_global.php on line 14

`define("LAN_PLUGIN_HERO_SUMMARY", "Gestion de la zone "Hero" de la page d'accueil");`
(double quote around Hero) 

and another one 
This one required some investigation ...
> PHP Fatal error:  Uncaught Error: Undefined constant "PAGE_NAME" in /home/ericc/www/ericc/e107_admin/cron.php:741\nStack trace:\n#0 {main}\n  thrown in /home/ericc/www/ericc/e107_admin/cron.php on line 741

in fact come from /home/ericc/www/ericc/e107_languages/French/admin/lan_cron.php, missing :
`if (!defined("PAGE_NAME")) { define("PAGE_NAME", "Planification des tâches"); }`

**Reply by tgtje** [reply]
thank you @ericc-dream !  1 was untouched (cron) ; Hero ?? anyway i already submitted changes, much appreciated.
